File: src/main/java/abstract_factory/product/UltraTruck.java
package abstract_factory.product;

import abstract_factory.product.part.VehiclePartFactory;

import java.util.List;

public class UltraTruck extends Vehicle {
    public static final String NAME = "Ultra Truck";
    private final VehiclePartFactory vehiclePartFactory;

    public UltraTruck(VehiclePartFactory vehiclePartFactory) {
        name = NAME;
        this.vehiclePartFactory = vehiclePartFactory;
    }

    @Override
    public void assembly() {
        engine = vehiclePartFactory.createEngine();
        tire = vehiclePartFactory.createTire();
        options = List.of();
    }
}

File: typescript/abstract-factory/product/ultra-truck.ts
import { VehiclePartFactory } from "./part/vehicle_part_factory";
import { Vehicle } from "./vehicle";

export class UltraTruck extends Vehicle {
    static readonly NAME = 'Ultra Truck';

    constructor(private readonly vehiclePartFactory: VehiclePartFactory) {
        super();
        this.name = UltraTruck.NAME;
    }

    assembly() {
        this.engine = this.vehiclePartFactory.createEngine();
        this.tire = this.vehiclePartFactory.createTire();
        this.options = new Array();
    }  
}